### ISSUE TITLE
Epoch issues

### ISSUE THREAD

**Issue by aggieNick02** [axboe/fio#1544]
**Please acknowledge the following before creating a ticket**

- [X] I have read the GitHub issues section of [REPORTING-BUGS](../blob/master/REPORTING-BUGS).

**Description of the bug:**
By default, fio log files use timestamps based on an epoch that is the beginning of the job. If you have multiple jobs, especially with some jobs waiting on other jobs before starting, this means you have different epochs and the times in the logs for each job cannot be compared/ordered.

Using `log_alternate_epoch` solves this problem as the logs all then share a common epoch. Unfortunately, it introduces a new problem. That is, the start time for each job relative to this common epoch is not known. So the gap between a job starting and the first log entry is indeterminate (although typically small, 0-1ms). (Less critically, `log_alternate_epoch` also increases log file size because the number of characters for each log entry timestamp increases.)

The shortcomings of each approach can be resolved by using the beginning of job epoch for log files but recording the start time of each job against a common (alternate) epoch.

I put something like this together in this PR from a while back: https://github.com/axboe/fio/pull/1353/commits

Back then I was more concerned about correlating log entries against other events on the same machine, and the PR was aimed at simplifying log files while still being able to know time against a machine-based epoch. It was nice to have but not super important (imho).

But as I've started to work with fio experiments with multiple jobs, having timekeeping shortcomings both with the normal epoch and `log_alternate_epoch` has led me to the same solution (the PR) to a different problem.

I was hoping the PR could be reconsidered or maybe some other solution brainstormed? I'm happy to implement a solution, whether it looks like the PR or something completely different.
**Environment**: 
All

**fio version**:
All

**Reproduction steps**
A simple fio run with per sample logging, each job waiting on the previous, illustrates the issues

**Comment by aggieNick02** [comment]
Suggested modified approach is up: https://github.com/axboe/fio/pull/1617
